htlcswitch/link.go

```diff
@@ -2437,7 +2485,46 @@ func (l *channelLink) processRemoteAdds(fwdPkg *channeldb.FwdPkg,
 				continue
 			}
 
+			// We don't have the pre-image yet, but will hold onto
+			// the HTLC as forward it to the switch as if there're
+			// additional hops. We use a special delayed resolution
+			// channelID, and then use the HTLC ID of the add index
+			// for the original invoice.
 			preimage := invoice.Terms.PaymentPreimage
+			if !bytes.Equal(preimage[:], unknownPreimage[:]) {
+				err = l.cfg.Registry.AcceptInvoice(
```

Comment:
@cfromknecht, I worked your PoC into this PR. You wrote:

>one thing we may consider is logically grouping`switchSettledPackets` from switchPackets. both can be forwarded in a single call to forwardBatch, but we should post-process accepting the switchSettledPackets after they have been forwarded. rationale here is that AcceptInvoice will fire a notification, which if instantly acted upon, could result in the circuit not being opened when the switch tries to lookup the circuit
delaying until after the circuits have been atomically opened and committed will ensure this case doesn't arise

Is this still how we want to do this? Or is it better to make the `AcceptInvoice` call inside the switch?